src/libraries/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs

```diff
@@ -11,29 +11,29 @@ internal static class KnownHeaders
     {
         // If you add a new entry here, you need to add it to TryGetKnownHeader below as well.
 
-        public static readonly KnownHeader Accept = new KnownHeader("Accept", HttpHeaderType.Request, MediaTypeHeaderParser.MultipleValuesParser, null, StaticTable.Accept);
-        public static readonly KnownHeader AcceptCharset = new KnownHeader("Accept-Charset", HttpHeaderType.Request, GenericHeaderParser.MultipleValueStringWithQualityParser, null, StaticTable.AcceptCharset);
-        public static readonly KnownHeader AcceptEncoding = new KnownHeader("Accept-Encoding", HttpHeaderType.Request, GenericHeaderParser.MultipleValueStringWithQualityParser, null, StaticTable.AcceptEncoding);
-        public static readonly KnownHeader AcceptLanguage = new KnownHeader("Accept-Language", HttpHeaderType.Request, GenericHeaderParser.MultipleValueStringWithQualityParser, null, StaticTable.AcceptLanguage);
-        public static readonly KnownHeader AcceptPatch = new KnownHeader("Accept-Patch");
-        public static readonly KnownHeader AcceptRanges = new KnownHeader("Accept-Ranges", HttpHeaderType.Response, GenericHeaderParser.TokenListParser, null, StaticTable.AcceptRanges);
+        public static readonly KnownHeader Accept = new KnownHeader("Accept", HttpHeaderType.Request | HttpHeaderType.NonTrailing, MediaTypeHeaderParser.MultipleValuesParser, null, StaticTable.Accept);
```

Comment:
Yes, I doesn't list these headers explicitly, but it lists header classes. The above Accept headers either determine how to process the payload or control message framing.

> A sender MUST NOT generate a trailer that contains a field necessary
   for message framing (e.g., Transfer-Encoding and Content-Length),
   routing (e.g., Host), request modifiers (e.g., controls and
   conditionals in Section 5 of [RFC7231]), authentication (e.g., see
   [RFC7235] and [RFC6265]), response control data (e.g., see Section
   7.1 of [RFC7231]), or determining how to process the payload (e.g.,
   Content-Encoding, Content-Type, Content-Range, and Trailer).

Reply:
I think you're over-generalizing, the RFC links to specific lists for each category. The Accept* headers only advertise the capabilities of the client or server.

Content-Disposition and Content-Location I could see an argument for, but they're not directly called out and they are significantly different from the ones that are (Content-Length, Content-Type).

From? I don't see any reason to block this one.

Reply:
> I think you're over-generalizing, the RFC links to specific lists for each category.

These lists seem to be open-ended because in all cases they use **e.g.**

> response control data **(e.g.**, see Section 7.1 of [RFC7231])

Reply:
True, but I don't see how the Allow* or From headers match any of the blocked categories.

We need to be cautious about what gets blocked or we risk blocking real customer scenarios.

Reply:
I suggest we don't do this PR unless we get customer feedback.

Reply:
@davidfowl note some of the headers in this PR should be blocked:
ProxyAuthenticate, ProxyAuthorization, SetCookie2, WWWAuthenticate

Reply:
>@davidsh note some of the headers in this PR should be blocked

Ok. Then we should constrain as little as possible where it is obvious that the headers should be blocked. But I agree that doing more (where it's uncertain) risks customer compat issues.

Reply:
Ok, that's a valid point. I will revert everything but the most obvious cases mentioned above.

Reply:
I reverted Accept*, Allow and From. Please, check the rest. I kept Content-Disposition and Content-Location because the spec says

>determining how to process the payload (e.g., Content-Encoding, Content-Type, Content-Range, and Trailer)

Please let me now if you still have any objections.